cmd/share-createlink.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2017 Dropbox, Inc.
+// Author: Daniel Porteous
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/dropbox/sharing"
+	"github.com/spf13/cobra"
+)
+
+/**
+** Try to get the share link for a file if it already exists.
+** If it doesn't make a new share link for it.
+ */
+func getShareLink(cmd *cobra.Command, args []string) (err error) {
+	if len(args) != 1 {
+		printShareLinkUsage()
+		return
+	}
+
+	dbx := sharing.New(config)
+	path, err := filepath.Abs(args[0])
+	if err != nil {
+		return
+	}
+
+	// Confirm that the file exists.
+	exists, err := exists(path)
+	if !exists || err != nil {
+		print("The file / folder specified does not exist.\n")
+		return
+	}
+
+	// Try to get a link if it already exists.
+	if getExistingLink(dbx, path) {
+		return
+	}
+
+	// print("File / folder does not yet have a sharelink, creating one...\n")
+
+	// The file had no share link, let's get it.
+	getNewLink(dbx, path)
+
+	return
+}
+
+func printShareLinkUsage() {
+	fmt.Printf("Usage: %s share getlink [file / folder path]\n", os.Args[0])
+}
+
+/*
+** Try to get an existing share link for a file / folder.
+** It returns true if the file / folder had a link. Otherwise it returns false.
+ */
+func getExistingLink(dbx sharing.Client, path string) bool {
```

Comment:
The name of this function is counter-intuitive, as it doesn't actually return the existing link. Perhaps `checkExistingLink`?

Reply:
It doesn't just check the existing link though, it also returns it if it exists. I think `checkExistingLinkAndReturnIfExists` would be a bit wordy 😉. I can change it regardless if you want.